### PR TITLE
CI: update precommit & apply

### PR DESCRIPTION
**Pull request by Borda**
running `pre-cimmit autopudate` and apply new versions plus a few manual fixed asked by flake8

cc: @EnricoMi

### CODE REVIEW

**Comment by Borda** [comment]
@sfdye @EnricoMi could you pls have look / review :flamingo:

**Comment by Borda** [comment]
@EnricoMi mind have a look, pls :pray:

**Comment by Borda** [comment]
@EnricoMi, how is it going? :chipmunk:

**Comment by Borda** [comment]
@EnricoMi is there anything I can do to help it land? 🦩

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,35 +1,42 @@
...
   - repo: https://github.com/pre-commit/pre-commit-hooks
     rev: v4.4.0
     hooks:
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
       - id: check-symlinks
       - id: destroyed-symlinks
       - id: mixed-line-ending
-        args: [ --fix=lf ]
+        args: [ "--fix=lf" ]
```
Comment (EnricoMi) [inline .pre-commit-config.yaml:13]:
```suggestion
        args:
          - "--fix=lf"
```

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,35 +1,42 @@
...
 
   - repo: https://github.com/asottile/pyupgrade
-    rev: v3.1.0
+    rev: v3.10.1
     hooks:
       - id: pyupgrade
-        args:
-          - --py37-plus
+        args: [ "--py37-plus" ]
```
Comment (EnricoMi) [inline .pre-commit-config.yaml:19]:
```suggestion
        args:
          - "--py37-plus"
```

**Review by EnricoMi — CHANGES_REQUESTED** [review]
Thanks for tidying this up, few minor comments.

**Comment by EnricoMi** [comment]
Looks like there are some uncommitted changes: https://github.com/PyGithub/PyGithub/actions/runs/6867963961/job/18677425814?pr=2600#step:4:132

**Comment by Borda** [comment]
> Looks like there are some uncommitted changes: [PyGithub/PyGithub/actions/runs/6867963961/job/18677425814?pr=2600#step:4:132](https://github.com/PyGithub/PyGithub/actions/runs/6867963961/job/18677425814?pr=2600#step:4:132)

yeah, this would be normally resolved wit the bot installed...
so let me run it locally and push update...

**Review by EnricoMi — APPROVED** [review]
LGTM!

**Pull request merged**